libraries/ruby_install.rb

```diff
@@ -0,0 +1,247 @@
+      execute.command(install_command)
+      execute.environment(new_resource.environment)
+      execute.run_action(:run)
+    end
+
+    # Check if the given Ruby is installed by directory. If the full directory
+    # does not exist, we can do partial global matching.
+    #
+    # @return [true, false]
+    def installed?
+      ::File.directory?("/opt/rubies/ruby-#{version}") ||
+        Dir['/opt/rubies/ruby-*'].any? { |ruby| ruby.include?(version) }
+    end
+
+    def install_dependencies
+      # install build-essentials
+      package = Chef::Resource::Package.new('build-essential', run_context)
+      package.package_name('build-essential')
+      package.run_action(:install)
+
+      # install ruby-install
+      return if Chef::Sugar::Shell.installed_at_version?('ruby-install', '0.4.1')
+      ruby_install = Chef::Resource::RemoteInstall.new('ruby-install', run_context)
+      ruby_install.source('https://codeload.github.com/postmodern/ruby-install/tar.gz/v0.4.1')
+      ruby_install.version('0.4.1')
+      ruby_install.checksum('1b35d2b6dbc1e75f03fff4e8521cab72a51ad67e32afd135ddc4532f443b730e')
+      ruby_install.install_command('make install')
+      ruby_install.run_action(:install)
+    end
+  end
+
+  class Provider::RubyInstallWindows < Provider::LWRPBase
+    include Languages::Helper
+
+    provides :ruby_install, platform_family: 'windows'
+
+    def whyrun_supported?
+      true
+    end
+
+    action(:install) do
+      if installed?
+        Chef::Log.debug("#{new_resource} installed - skipping")
+      else
+        converge_by("install #{new_resource}") do
+          install_ruby
+          install_devkit
+          configure_ca
+        end
+      end
+    end
+
+    protected
+
+    def version
+      new_resource.version
+    end
+
+    def installer_url
+      "http://dl.bintray.com/oneclick/rubyinstaller/rubyinstaller-#{version}.exe"
+    end
+
+    def installer_download_path
+      windows_safe_path_join(Chef::Config[:file_cache_path], ::File.basename(installer_url))
+    end
+
+    # Determines the proper version of the DevKit based on Ruby version.
+    def devkit_url
+      # 2.0 64-bit
+      if version =~ /^2\.\d\.\d.*x64$/
+        'http://cdn.rubyinstaller.org/archives/devkits/DevKit-mingw64-64-4.7.2-20130224-1432-sfx.exe'
+      # 2.0 32-bit
+      elsif version =~ /^2\.\d\.\d.*$/
+        'http://cdn.rubyinstaller.org/archives/devkits/DevKit-mingw64-32-4.7.2-20130224-1151-sfx.exe'
+      # Ruby 1.8.7 and 1.9.3
+      else
+        'https://github.com/downloads/oneclick/rubyinstaller/DevKit-tdm-32-4.5.2-20111229-1559-sfx.exe'
+      end
+    end
+
+    def devkit_download_path
+      windows_safe_path_join(Chef::Config[:file_cache_path], ::File.basename(devkit_url))
+    end
+
+    def cacerts_download_path
+      windows_safe_path_join(Chef::Config[:file_cache_path], ::File.basename(devkit_url))
+    end
+
+    def ruby_install_path
+      windows_safe_path_join(ENV['SYSTEMDRIVE'], 'rubies', version)
```

Comment:
This should just be `new_resource.prefix` assuming you add the `prefix` attribute [mentioned above](https://github.com/chef-cookbooks/languages/pull/4/files#r42443869)